### DISCUSSION TITLE
Possible to create a standalone "Offline" option

### DISCUSSION THREAD

**Discussion by tcl301** [viliusle/miniPaint#409] [Ideas]
Have you considered creating a stand-alone HTML file with all of the resources embedded in the code?  I have attempted to do this with your APP but have not been successful to date.  In some cases, internet access isn't always available nor is the ability to setup NPM and all of the trimmings that go with it.  Otherwise, thanks for the APP I use it everyday.

**Comment by M1FSS** [comment]
https://github.com/viliusle/miniPaint/wiki/Docker

**Reply by tcl301** [reply]
Thanks for the docker solutions but still requires external software other than a browser.  Assume ALL I have access to is a web browser, no docker, no git.  Understood that if updates are needed, the standalone HTML file would have to be rebuilt.

**Reply by tcl301** [reply]
While I haven't tried it, probably something like this: https://github.com/zTrix/webpage2html. I doubt that it will work with miniPaint since I've tried manually in the past to do the same thing and clearly miniPaint is not a simple "HTML page".

**Comment by viliusle** [comment]
1. go to [github](https://github.com/viliusle/miniPaint)
2. choose master or live (gh-pages) repository
3. download files (code button > download ZIP)
4. extract archive, lets say Desktop
5. open index.html file with browser and here you will be able to use app without internet access.

p.s. bookmark opened file for future use
p.s.s. time to time you should check if there is new version on https://viliusle.github.io/miniPaint/

**Comment by tcl301** [comment]
Thank you .. guess I should have known to look there .. I'll let you know if I'm able to base64 the fonts and images and wrap them all up in the single index page to create one single "BrowserApp". I will not release to the general population.

This might seem like a strange request but in environments where the internet is not available and your average user does not have access to create an 8080 port or their own server, this ends up being the only option. It has worked well in the past, I could send you some examples of other WebApps I've turned into BrowserApps if you're interested. Yours will be the most complicated one I've converted however. You've given me half the battle though.

Tad Lewis

**Reply by FraterCCCLXIII** [reply]
Have you considered wrapping it in electron as a solution?
